src/zenml/integrations/mlflow/experiment_trackers/mlflow_experiment_tracker.py

```diff
@@ -0,0 +1,185 @@
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import os
+from typing import Any, ClassVar, Dict, Optional
+
+from mlflow import (  # type: ignore[import]
+    ActiveRun,
+    get_experiment_by_name,
+    search_runs,
+    set_experiment,
+    set_tracking_uri,
+    start_run,
+)
+from mlflow.entities import Experiment  # type: ignore[import]
+from mlflow.store.db.db_types import DATABASE_ENGINES
+from pydantic import root_validator, validator
+
+from zenml.experiment_trackers.base_experiment_tracker import (
+    BaseExperimentTracker,
+)
+from zenml.integrations.constants import MLFLOW
+from zenml.logger import get_logger
+from zenml.repository import Repository
+from zenml.stack.stack_component_class_registry import (
+    register_stack_component_class,
+)
+
+logger = get_logger(__name__)
+
+MLFLOW_TRACKING_USERNAME = "MLFLOW_TRACKING_USERNAME"
+MLFLOW_TRACKING_PASSWORD = "MLFLOW_TRACKING_PASSWORD"
+MLFLOW_TRACKING_TOKEN = "MLFLOW_TRACKING_TOKEN"
+MLFLOW_TRACKING_INSECURE_TLS = "MLFLOW_TRACKING_INSECURE_TLS"
+
+
+# Add validation
+def _local_mlflow_backend() -> str:
+    """Returns the local mlflow backend inside the zenml artifact
+    repository directory
+
+    Returns:
+        The MLflow tracking URI for the local mlflow backend.
+    """
+    repo = Repository(skip_repository_check=True)  # type: ignore[call-arg]
+    artifact_store = repo.active_stack.artifact_store
+    local_mlflow_backend_uri = os.path.join(artifact_store.path, "mlruns")
+    if not os.path.exists(local_mlflow_backend_uri):
+        os.makedirs(local_mlflow_backend_uri)
+        # TODO [MEDIUM]: safely access (possibly non-existent) artifact stores
+    return "file:" + local_mlflow_backend_uri
+
+
+@register_stack_component_class
+class MLFlowExperimentTracker(BaseExperimentTracker):
+    """Configure the MLFlow Experiment Tracker.
+
+    Manages the global MLflow environment in the form of a Stack
+    component. To access it inside your step function or in the post-execution
+    workflow:
+
+    ```python
+    from zenml.steps import StepContext
+
+    @enable_mlflow
+    @step
+    def my_step(context: StepContext, ...)
+        context.stack.experiment_tracker  # This is the MLFlowExperimentTracker object
+    ```
+    """
+
+    tracking_uri: Optional[str] = None
+    tracking_username: Optional[str] = None
+    tracking_password: Optional[str] = None
+    tracking_token: Optional[str] = None
+    tracking_insecure_tls: bool = False
+
```

Comment:
The MLflow client might also need additional credentials to access the artifact store directly (see [scenario 4](https://www.mlflow.org/docs/latest/tracking.html#scenario-4-mlflow-with-remote-tracking-server-backend-and-artifact-stores)). It is also true that these credentials can also be supplied at pipeline level instead of being configured in the stack component. However, specifying them in the stack component (e.g. like a reference to a ZenML secret, same as I did with Seldon Core) has the advantage that you don't need to change the pipeline code when you switch the MLflow tracking service.